script/setup_local_dev.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+
+: "${ENSURE_K3D:="false"}" # will ensure k3d is installed if set to "true"
+: "${ENSURE_KUBECTL:="false"}" # will ensure kubectl is installed if set to "true"
+: "${ENSURE_HELM:="false"}" # will ensure helm is installed if set to "true"
+
+: "${K3D_VERSION:="v4.4.8"}" # version of k3d to install, empty value uses latest one available. currently pinned to latest known working (due to unresolved port mapping issue: https://github.com/k3d-io/k3d/issues/380)
```

Comment:
So I updated to using 5.4.1 with the following command:
`k3d cluster create -p "30082:30082@server:0" -p "30084:30084@server:0" -p "30089:30089@server:0" --image rancher/k3s:v1.21.8-k3s2 -v /dev/mapper/nvme0n1p3_crypt:/dev/mapper/nvme0n1p3_crypt flyte`

I think if we just use a single server with no agents and map our forwarded ports to the server instance there is no port mapping issue? Or is this a load-balancer issue? Not a high priority change, but it's nice to use more recent software.

Reply:
I have admittedly not tried using newer versions as the old one _just worked_. Will try out the newer version this week to see if I run into any issues - otherwise I'll bump the version as suggested :+1:

Reply:
Seems to be working fine for all tests I've done locally, the only downside is running traefik/a loadbalancer although not strictly necessary. I'm fine with that though, since the impact should be non-existent anyways.
I agree that being able to use a newer version of k3d would be preferable :+1: